src/esp/gfx_batch/Renderer.cpp

```diff
@@ -381,29 +502,33 @@ void Renderer::addFile(const Cr::Containers::StringView filename,
        the same mapping. */
     const Cr::Containers::StridedArrayView1D<const Mn::UnsignedInt>
         meshViewMapping =
-            scene->mapping<Mn::UnsignedInt>(*meshViewIndexCountFieldId);
+            scene->mapping<Mn::UnsignedInt>(Mn::Trade::SceneField::Mesh);
     for (std::size_t i = 0; i != meshViewMapping.size(); ++i) {
-      state_->meshViews[i].transformation = transformations[meshViewMapping[i]];
+      meshViews[i].transformation = transformations[meshViewMapping[i]];
     }
 
     /* Templates are the root objects with their names. Their immediate
        children are the actual meshes. Assumes the order matches the order of
        the custom fields. */
     // TODO hacky and brittle! doesn't handle nested children properly, doesn't
     //  account for a different order of the field vs the child lists
-    Mn::UnsignedInt offset = 0;
+    Mn::UnsignedInt offset = meshViewOffset;
     for (Mn::UnsignedLong root : scene->childrenFor(-1)) {
       Cr::Containers::Array<Mn::UnsignedLong> children =
           scene->childrenFor(root);
 
       Cr::Containers::String name = importer->objectName(root);
       CORRADE_ASSERT(name,
                      "Renderer::addFile(): node" << root << "has no name", );
-      state_->meshViewRangeForName.insert(
-          {name, {offset, offset + Mn::UnsignedInt(children.size())}});
+      CORRADE_ASSERT_OUTPUT(
```

Comment:
Hm, for this one I actually got an idea -- would it make sense that if you `addFile()` that contains  identifiers/names that are already present from files added earlier, it replaces them? Instead of failing/asserting. 

Let's say, for iterative workflows where you have some simulation running already and just need to figure the best scale/color/whatever for a newly added object. Or for lights. There it might make sense that you repeatedly `addFile()` and `addMeshHierarchy()` until you're happy. The consequence is of course that you gradually fill GPU memory with unused meshes, but might be an okay tradeoff compared to having to suffer the delay of shutting down and restarting everything. Or it could be an opt-in flag in the config, to allow this, and be strict by default.

Thoughts?

Reply:
Yeah, hard to guess what users want here, but opt-in flag with strict by default sounds good.

Reply:
Done as well, this is now a runtime error and the function will return `false` in that case. 

To avoid too many unused features creeping in, I didn't add the opt-in flag to replace existing names. There's a TODO and I will implement it once an actual use case appears.